library/std/src/io/buffered/bufreader/buffer.rs

```diff
@@ -0,0 +1,84 @@
+use crate::cmp;
+use crate::io::{self, Read, ReadBuf};
+use crate::mem::MaybeUninit;
+
+pub struct Buffer {
+    buf: Box<[MaybeUninit<u8>]>,
+    pos: usize,
+    cap: usize,
+    init: usize,
```

Comment:
It'd be great to document these fields. `cap` in a particular is a little confusing, since it's not quite the capacity -- which would be buf.len() -- of the storage.